components/forward_post_modal/forward_post_modal.tsx

```diff
@@ -0,0 +1,273 @@
+                    values={{
+                        participants: <FormattedList value={participants}/>,
+                    }}
+                />
+            );
+        }
+
+        notification = (
+            <NotificationBox
+                variant={'info'}
+                text={notificationText}
+            />
+        );
+    }
+
+    const handleSubmit = useCallback(async () => {
+        let result = await actions.forwardPost(
+            post,
+            isPrivateConversation ? currentChannel.id : selectedChannelId,
+            comment,
+        );
+        if (result.error) {
+            setPostError(result.error);
+            return;
+        }
+
+        if (
+            selectedChannel?.details.type === Constants.MENTION_MORE_CHANNELS &&
+            selectedChannel?.details.type === Constants.OPEN_CHANNEL
+        ) {
+            result = await actions.joinChannelById(selectedChannelId);
+
+            if (result.error) {
+                setPostError(result.error);
+                return;
+            }
+        }
+
+        // only switch channels when we are not in a private conversation
+        if (!isPrivateConversation && selectedChannel) {
+            result = await actions.switchToChannel(selectedChannel.details);
+
+            if (result.error) {
+                setPostError(result.error);
+                return;
+            }
+        }
+
+        onHide();
+    }, [
+        actions,
+        post,
+        isPrivateConversation,
+        currentChannel.id,
+        selectedChannelId,
+        comment,
+        selectedChannel,
+        onHide,
+    ]);
+
+    const postPreviewFooterMessage = formatMessage({
+        id: 'forward_post_modal.preview.footer_message',
+        defaultMessage: 'Originally posted in ~{channelName}',
+    },
+    {
+        channel: currentChannel.display_name,
+    });
+
+    const providerValue = {
+        comment,
+        setComment,
+        setPostError,
+    };
+
+    return (
+        <GenericModal
+            className='a11y__modal forward-post'
+            show={true}
+            enforceFocus={false}
+            autoCloseOnConfirmButton={false}
+            useCompassDesign={true}
+            modalHeaderText={formatMessage({
+                id: 'forward_post_modal.title',
+                defaultMessage: 'Forward message',
+            })}
+            confirmButtonText={formatMessage({
+                id: 'forward_post_modal.button.forward',
+                defaultMessage: 'Forward',
+            })}
+            cancelButtonText={formatMessage({
+                id: 'forward_post_modal.button.cancel',
+                defaultMessage: 'Cancel',
+            })}
+            isConfirmDisabled={!canForwardPost}
+            handleConfirm={handleSubmit}
+            handleEnterKeyPress={handleSubmit}
+            handleCancel={onHide}
+            onExited={onHide}
+        >
+            <ForwardPostContext.Provider value={providerValue}>
```

Comment:
my original idea was to show the text together with the message preview ... like it would be shown when viewed in the channel. I have misread the requirements in that case though, since it was not really wanted and produced more problems than it solved.

I never got to change it back since it works that way perfectly fine ... I am not against changing that to passing down props.